packages/components/src/input-control/input-field.tsx

```diff
@@ -116,14 +84,14 @@ function InputField(
 			if ( ! isValueEmpty( value ) ) {
 				handleOnCommit( event );
 			} else {
-				reset( valueProp, event );
+				reset( undefined, event );
 			}
 		}
 	};
 
 	const handleOnFocus = ( event: FocusEvent< HTMLInputElement > ) => {
 		onFocus( event );
-		setIsFocused?.( true );
+		update( { isFocused: true, initialValue: value } );
```

Comment:
The reset action will use the `initialValue` from state if its first argument is falsy. Now the `initialValue` is updated upon `focus` and it seems easier to reason about that way instead of passing a value from props to `reset`. It's also not clear to me that it worked as intended before but I suppose this is one part to double-check if tests exist and serve their intended purpose.